x-pack/legacy/plugins/snapshot_restore/public/app/components/restore_snapshot_form/steps/step_logistics.tsx

```diff
@@ -156,89 +174,158 @@ export const RestoreSnapshotStepLogistics: React.FunctionComponent<StepProps> =
+                        values={{
+                          count: restoreIndices && restoreIndices.length,
+                          selectOrDeselectAllLink:
+                            restoreIndices && restoreIndices.length > 0 ? (
+                              <EuiLink
+                                onClick={() => {
+                                  indicesOptions.forEach((option: Option) => {
+                                    option.checked = undefined;
+                                  });
+                                  updateRestoreSettings({ indices: [] });
+                                  setCachedRestoreSettings({
+                                    ...cachedRestoreSettings,
+                                    indices: [],
+                                  });
+                                }}
+                              >
+                                <FormattedMessage
+                                  id="xpack.snapshotRestore.restoreForm.stepLogistics.deselectAllIndicesLink"
+                                  defaultMessage="Deselect all"
+                                />
+                              </EuiLink>
+                            ) : (
+                              <EuiLink
+                                onClick={() => {
+                                  indicesOptions.forEach((option: Option) => {
+                                    option.checked = 'on';
+                                  });
+                                  updateRestoreSettings({ indices: [...snapshotIndices] });
+                                  setCachedRestoreSettings({
+                                    ...cachedRestoreSettings,
+                                    indices: [...snapshotIndices],
+                                  });
+                                }}
+                              >
+                                <FormattedMessage
+                                  id="xpack.snapshotRestore.restoreForm.stepLogistics.selectAllIndicesLink"
+                                  defaultMessage="Select all"
+                                />
+                              </EuiLink>
+                            ),
+                        }}
+                      />
+                    ) : null
                   }
                   isInvalid={Boolean(errors.indices)}
                   error={errors.indices}
                 >
-                  <EuiSelectable
-                    allowExclusions={false}
-                    options={indicesOptions}
-                    onChange={options => {
-                      const newSelectedIndices: string[] = [];
-                      options.forEach(({ label, checked }) => {
-                        if (checked === 'on') {
-                          newSelectedIndices.push(label);
+                  {selectIndicesMode === 'list' ? (
```

Comment:
Minor nit: we have three parallel branches that depend upon `selectIndicesMode`, do you think it would make sense to reduce this to a single branch that switches between an `IndicesSelect` component and an `IndexPatternInput` component? That would also reduce the size of this file, which is getting lengthy.